Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -5334,102 +5350,63 @@ bool CSSPropertyParser::consumeFont(bool important)
-            if (!fontVariantCaps && (fontVariantCaps = consumeIdent<CSSValueSmallCaps>(range)))
-                continue;
-            if (!fontWeight && (fontWeight = consumeFontWeight(range)))
-                continue;
-            if (!fontStretch && (fontStretch = consumeFontStretchKeywordValue(range, CSSValuePool::singleton())))
-                continue;
-            break;
-        }
+    // Optional font-style, font-variant, font-stretch and font-weight, in any order.
+    for (unsigned i = 0; i < 4 && !range.atEnd(); ++i) {
+        if (consumeIdent<CSSValueNormal>(range))
+            continue;
+        if (!fontStyle && (fontStyle = consumeFontStyle(range, m_context.mode, CSSValuePool::singleton())))
+            continue;
+        if (!fontVariantCaps && (fontVariantCaps = consumeIdent<CSSValueSmallCaps>(range)))
+            continue;
+        if (!fontWeight && (fontWeight = consumeFontWeight(range)))
+            continue;
+        if (!fontStretch && (fontStretch = consumeFontStretchKeywordValue(range, CSSValuePool::singleton())))
+            continue;
+        break;
+    }
 
-        if (range.atEnd())
-            return false;
+    if (range.atEnd())
+        return false;
 
-        fontSize = consumeFontSize(range, m_context.mode);
-        if (!fontSize || range.atEnd())
-            return false;
+    fontSize = consumeFontSize(range, m_context.mode);
+    if (!fontSize || range.atEnd())
+        return false;
 
-        if (consumeSlashIncludingWhitespace(range)) {
-            if (!consumeIdent<CSSValueNormal>(range)) {
-                lineHeight = consumeLineHeight(range, m_context.mode);
-                if (!lineHeight)
-                    return false;
-            }
+    if (consumeSlashIncludingWhitespace(range)) {
+        if (!consumeIdent<CSSValueNormal>(range)) {
+            lineHeight = consumeLineHeight(range, m_context.mode);
+            if (!lineHeight)
+                return false;
         }
-
         if (range.atEnd())
             return false;
-
-        fontFamily = consumeFontFamily(range);
-        if (!fontFamily)
-            return false;
     }
 
-    if (!range.atEnd())
+    fontFamily = consumeFontFamily(range);
+    if (!fontFamily || !range.atEnd())
         return false;
 
-    auto reset = [&] (CSSPropertyID property) {
-        addProperty(property, CSSPropertyFont, CSSValuePool::singleton().createImplicitInitialValue(), important, true);
+    auto reset = [&] (CSSPropertyID property, CSSValueID initialValue) {
+        ASSERT(initialValue != CSSValueInvalid);
+        addProperty(property, CSSPropertyFont, CSSValuePool::singleton().createIdentifierValue(initialValue), important, true);
     };
     auto add = [&] (CSSPropertyID property, RefPtr<CSSValue>& value) {
         if (value)
             addProperty(property, CSSPropertyFont, value.releaseNonNull(), important);
         else
-            reset(property);
+            reset(property, CSSValueNormal);
     };
```

Comment:
This `add` function is very similar to `addPropertyWithImplicitDefault`.

I think that method could be used directly, and `addProperty` could be used for `font-size`/`font-family` (since they are mandatory).

Or if you want to avoid the verbosity, you could also rewrite `add` to be:
```
auto add = [&] (CSSPropertyID property, RefPtr<CSSValue>& value) {
    if (property == CSSPropertyFontSize || property == CSSPropertyFontFamily)
        addProperty(property, CSSPropertyFont, value.releaseNonNull(), important);
    else
        addPropertyWithImplicitDefault(property, CSSPropertyFont, value.releaseNonNull(), CSSValuePool::singleton().createIdentifierValue(CSSValueNormal), important);
};
```

Reply:
(could also be done separately if you prefer)